Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,305 @@
+        }
+
+        private static List<JObject> GetSchemaFromPath(string path, out List<string> errorList, bool validateAsDictionary)
+        {
+            errorList = new List<string>();
+
+            try
+            {
+                string schema = File.ReadAllText(path);
+
+                List<JObject> res = ConvertStringToJObjectList(schema, out List<string> convertError, validateAsDictionary);
+                errorList = convertError;
+
+                return res;
+            }
+            catch (Exception e)
+            {
+                errorList.Add(e.Message);
+
+                return new List<JObject>();
+            }
+        }
+
+        private static List<JObject> GetAllSchemasInDirectory(string path, out List<string> errorList, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+            errorList = new List<string>();
+
+            try
+            {
+                string[] Files = Directory.GetFiles(path);
+                List<object> schemalist = new List<object>();
+
+                if (validateAsDictionary) {
+                    Dictionary<string, ForgeTree> combinedDictionary = new Dictionary<string, ForgeTree>();
+                    
+                    foreach (string file in Files) 
+                    {
+                        string schema = File.ReadAllText(file);
+                        Dictionary<string, ForgeTree> schemaDictionary = JsonConvert.DeserializeObject<Dictionary<string, ForgeTree>>(schema);
+                        
+                        foreach (KeyValuePair<string, ForgeTree> item in schemaDictionary) 
+                        {
+                            combinedDictionary.Add(item.Key, item.Value);
+                        }
+                    }
+
+                    return new List<JObject> { SerializeToJObject(combinedDictionary) };
+                }
+                else
+                {
+                    foreach (string file in Files)
+                    {
+                        List<JObject> schemasInFile = GetSchemaFromPath(file, out errorList, validateAsDictionary);
+
+                        if (errorList.Count > 0)
+                        {
+                            break;
+                        }
+
+                        schemasInFile.ForEach(n => schemaList.Add(n));
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                errorList.Add(e.Message);
+            }
+
+            return schemaList;
+        }
+
+        private static bool CheckConvertErrorAndValidate(Object rules, ref List<string> errorList, List<JObject> schemaList)
+        {
+            if (errorList.Count > 0)
+            {
+                return false;
+            }
+
+            return Validate(schemaList, rules, out errorList);
+        }
+
+        private static bool Validate(List<JObject> schemas, Object rules, out List<string> errorList)
```

Comment:
>private static bool Validate [](http://example.com/codeflow?start=8&length=28)

I'm okay not adding summary tags to the other private methods in this class because they are self-explanatory. But let's add some richer documentation here since it covers many different combinations of inputs.

The goal is the have this code be maintainable. Imagine you are writing the documentation for someone that needs to understand it 2 years from now.